cluster-location-metadata/README.md

```diff
@@ -0,0 +1,332 @@
+
+Regardless of the environment (cloud/on-prem), the value of the `topology.kubernetes.io/region` label should not include any whitespace and consist of the lowercase letters, numbers, and dash only (`[a-z0-9-]`).
+
+#### Default location ID system for on-prem
+
+The metadata system described above allows to specify the location of a cluster as a single string.
+
+For our on-premises customers, we want to provide a default system to inidicate their cluster's locations. This default system should be
+
+- easy to use and to understand
+- hierarchical
+- adaptable/customizable
+
+It should be up to the customer to decide if they want to deviate from the system and populate the label mentioned above with whatever string they like. For example, let's just assume a customer has a numeric system to enumerate their facilities. In that case, they should be able to use that identifier instead of our default system.
+
+For our default system, here is a synopsis of the format:
+
+```
+<CONTINENT_CODE>-<COUNTRY_CODE>[-<SUBDIVISION_CODE>[-<CITY_NAME>]]
+```
+
+This means that the location ID string is composed of two to four components, separated by a dash. The first two parts, the continent code and the country code, are mandatory. If a location has to be specified more precisely, a state code can be added, and lastly a city could can be added for hightest precision.
+
+Depending on the customer's needs, a cluster in Frankfurt am Main, Germany, could bear either of the following three location identifiers:
+
+- Country-level precision: `eu-de`
+- Country subdivision level precision: `eu-de-he`
+- City level precision: `eu-de-he-frankfurt`
+
+Component details:
+
+- `CONTINENT_CODE`: The mandatory first part is a two-letter code for the global region/continent. Available are:
+  - `af` for Africa
+  - `an` for Antarctica (yes, unlikely, but who knows)
+  - `as` for Asia
+  - `eu` for Europe
+  - `na` for North America
+  - `oc` for Oceania
+  - `sa` for South America
+- `COUNTRY_CODE` is the two letter [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) code of the country the cluster is located in.
+- `SUBDIVISION_CODE` is the code for a subdivision of the country, like it might makes sense for larger countries like the U. S. If CITY_NAME is specified, the SUBDIVISION_CODE must be given.
+- `CITY_NAME` is a concise form of the city name the cluster is hosted in, also optional.
+
+### Alternative solutions
+
+- Instead of `topology.kubernetes.io/region`, we could specify our own label in the `giantswarm.io` namespace. The latter would make sense in case we wanted to mark that there would be some logical difference between these labels.
+
+- Instead of strings, we could introduce a system that provides geo coordinates. (Since the cloud providers already work with string identifiers, this draft opts for a solution that extends this paradigm to on-prem.)
```

Comment:
Something like [what 3 words](https://what3words.com/clip.apples.leap) could also work and maybe be human-friendlier.

However, the exact locations of some sites are secret, so we would definitely have some opt-outs from this approach.